torchvision/datasets/widerface.py

```diff
@@ -0,0 +1,231 @@
+from os.path import abspath, expanduser
+import torch
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from .utils import download_file_from_google_drive, check_integrity, download_url, extract_archive
+from .vision import VisionDataset
+
+
+class WIDERFace(VisionDataset):
+    """`WIDERFace <http://shuoyang1213.me/WIDERFACE/>`_ Dataset.
+
+    Citation:
+    @inproceedings{yang2016wider,
+        author    = "Yang, Shuo and Luo, Ping and Loy, Chen Change and Tang, Xiaoou",
+        booktitle = "IEEE Conference on Computer Vision and Pattern Recognition (CVPR)",
+        title     = "WIDER FACE: A Face Detection Benchmark",
+        year      = "2016"}
+
+    Args:
+        root (string): Root directory where images and annotations are downloaded to.
+            Expects the following folder structure if download=False:
+                <root>
+                    └── widerface
+                        ├── wider_face_split.zip
+                        ├── WIDER_train.zip
+                        ├── WIDER_val.zip
+                        └── WIDER_test.zip
+        split (string): The dataset split to use. One of {``train``, ``val``, ``test``}.
+            Defaults to ``train``.
+        target_type (string): The type of target to use, can be one
+            of {``raw``, ``bbox``, ``attr``.``""``}. Can also be a list to
+            output a tuple with all specified target types.
+            The targets represent:
+                ``raw``  (torch.tensor shape=(10,) dtype=int): all annotations combined (bbox + attr)
+                ``bbox`` (torch.tensor shape=(4,) dtype=int): bounding box (x, y, width, height)
+                ``attr`` (torch.tensor shape=(6,) dtype=int): label values for attributes
+                    that represent (blur, expression, illumination, occlusion, pose, invalid)
+            Defaults to ``raw``. If empty, ``None`` will be returned as target.
+        transform (callable, optional): A function/transform that  takes in a PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    """
+
+    base_folder = "widerface"
+    file_list = [
+        # File ID                        MD5 Hash                            Filename
+        ("0B6eKvaijfFUDQUUwd21EckhUbWs", "3fedf70df600953d25982bcd13d91ba2", "WIDER_train.zip"),
+        ("0B6eKvaijfFUDd3dIRmpvSk8tLUk", "dfa7d7e790efa35df3788964cf0bbaea", "WIDER_val.zip"),
+        ("0B6eKvaijfFUDbW4tdGpaYjgzZkU", "e5d8f4248ed24c334bbd12f49c29dd40", "WIDER_test.zip")
+    ]
+    annotations_file = ("http://mmlab.ie.cuhk.edu.hk/projects/WIDERFace/support/bbx_annotation/wider_face_split.zip",
```

Comment:
To indicate that these are constants, could you capitalize them?

Reply:
No  problem. I was following conventions found in other dataset classes, but capitalization makes sense here.

Reply:
We'll happily accept PRs changing that :wink: